airflow/api_fastapi/core_api/routes/ui/dags.py

```diff
@@ -0,0 +1,133 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from fastapi import Depends
+from sqlalchemy import and_, func, select
+from sqlalchemy.orm import Session
+from typing_extensions import Annotated
+
+from airflow.api_fastapi.common.db.common import (
+    get_session,
+    paginated_select,
+)
+from airflow.api_fastapi.common.parameters import (
+    QueryDagDisplayNamePatternSearch,
+    QueryDagIdPatternSearch,
+    QueryLastDagRunStateFilter,
+    QueryLimit,
+    QueryOffset,
+    QueryOnlyActiveFilter,
+    QueryOwnersFilter,
+    QueryPausedFilter,
+    QueryTagsFilter,
+)
+from airflow.api_fastapi.common.router import AirflowRouter
+from airflow.api_fastapi.core_api.serializers.dag_run import DAGRunResponse
+from airflow.api_fastapi.core_api.serializers.dags import DAGResponse
+from airflow.api_fastapi.core_api.serializers.ui.dags import (
+    DAGWithLatestDagRunsCollectionResponse,
+    DAGWithLatestDagRunsResponse,
+)
+from airflow.models import DagModel, DagRun
+
+dags_router = AirflowRouter(prefix="/dags", tags=["Dags"])
+
+
+@dags_router.get("/recent_dag_runs", include_in_schema=False, response_model_exclude_none=True)
+async def recent_dag_runs(
+    limit: QueryLimit,
+    offset: QueryOffset,
+    tags: QueryTagsFilter,
+    owners: QueryOwnersFilter,
+    dag_id_pattern: QueryDagIdPatternSearch,
+    dag_display_name_pattern: QueryDagDisplayNamePatternSearch,
+    only_active: QueryOnlyActiveFilter,
+    paused: QueryPausedFilter,
+    last_dag_run_state: QueryLastDagRunStateFilter,
+    session: Annotated[Session, Depends(get_session)],
+    dag_runs_limit: int = 10,
```

Comment:
This limit shouldn't the same as the `limit`. So we can specify `x` dags, but `y` dag runs per dag. I updated it accordingly.